test/shacl.test.ts

```diff
@@ -7,22 +7,34 @@ import DatasetExt from 'rdf-ext/lib/Dataset';
 import {DataFactory} from 'rdf-data-factory';
 
 describe('Dataset', () => {
-  it('validates against SHACL', async () => {
+  it('validates against SHACL', async done => {
     const datasets = listDatasets();
 
     const validator = await shaclValidator();
+    const jsonLdParser = new JsonLdParser();
+    const base = await factory
+      .dataset()
+      .import(
+        fs.createReadStream('catalog/publishers.jsonld').pipe(jsonLdParser)
+      );
+
     for (const dataset of datasets) {
       const jsonLdParser = new JsonLdParser();
-      const data = await factory
-        .dataset()
-        .import(
-          fs.createReadStream('catalog/datasets/' + dataset).pipe(jsonLdParser)
-        );
+      const data = (
+        await factory
+          .dataset()
+          .import(
+            fs
+              .createReadStream('catalog/datasets/' + dataset)
+              .pipe(jsonLdParser)
+          )
+      ).addAll(base);
       expect(containsDatasetNode(data)).toBe(true);
 
       const report = await validator.validate(data);
       expect(report).toConform();
     }
+    done();
```

Comment:
A callback, interesting! I'd say that it shouldn't be necessary: it's an async function. Or does the use of streams require the callback?

Reply:
I was running in some test timeout issues, so messed around with calling the callbacks to make sure no lingering async was keeping the test running. Removing it again, as it doesn’t really seem to change anything (and callbacks are evil 👹).